src/stdlib_experimental_io.f90

```diff
@@ -351,34 +344,44 @@ integer function open(filename, mode) result(u)
 open(newunit=u, file=filename, &
      action = action_, position = position_, status = status_, &
      access = access_, form = form_, &
-     iostat = io)
+     iostat = io_)
+
+if(present(io))io=io_
```

Comment:
If it is not present, then we now need to give an error. For the reason, I would probably let the compiler generate the proper error message, and write this like this:
```fortran
if (present(io)) then
    open(newunit=u, file=filename, &
        action = action_, position = position_, status = status_, &
        access = access_, form = form_, &
        iostat = io)
else
    open(newunit=u, file=filename, &
        action = action_, position = position_, status = status_, &
        access = access_, form = form_)
end if
```